packages/core/contracts/financial-templates/common/financial-product-libraries/KpiOptionsFinancialProductLibrary.sol

```diff
@@ -11,6 +11,42 @@ import "../../../common/implementation/Lockable.sol";
  * Post-expiry, the collateral requirement is left as 1 and the price is left unchanged.
  */
 contract KpiOptionsFinancialProductLibrary is FinancialProductLibrary, Lockable {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    mapping(address => FixedPoint.Unsigned) financialProductTransformedPrices;
+
+    /**
+     * @notice Enables any address to set the transformed price for an associated financial product.
+     * @param financialProduct address of the financial product.
+     * @param transformedPrice the price for the financial product to be used if the contract is pre-expiration.
+     * @dev Note: a) Any address can set price transformations b) The price can't be set to blank.
+     * c) A transformed price can only be set once to prevent the deployer from changing it after the fact.
+     * d)  financialProduct must expose an expirationTimestamp method.
+     */
+    function setFinancialProductTransformedPrice(address financialProduct, FixedPoint.Unsigned memory transformedPrice)
+        public
+        nonReentrant()
+    {
+        require(transformedPrice.isGreaterThan(0), "Cant set price of 0");
+        require(financialProductTransformedPrices[financialProduct].isEqual(0), "Price already set");
+        require(ExpiringContractInterface(financialProduct).expirationTimestamp() != 0, "Invalid EMP contract");
+        financialProductTransformedPrices[financialProduct] = transformedPrice;
+    }
+
+    /**
+     * @notice Returns the transformed price associated with a given financial product address.
+     * @param financialProduct address of the financial product.
+     * @return transformed price for the associated financial product.
+     */
+    function getTransformedPriceForFinancialProduct(address financialProduct)
+        public
+        view
+        nonReentrantView()
```

Comment:
Don't think you need this `nonReentrantView` since there is no external call

Reply:
is this function redundant if `financialProductTransformedPrices` map is public?

Reply:
Agreed @daywiss